app/src/main/java/pl/edu/pjatk/goodwill_ninjas/blooddonor_android/components/login/SignInScreen.kt

```diff
@@ -0,0 +1,158 @@
+    Column {
+        TextField(
+                modifier = Modifier.fillMaxWidth(),
+                value = email,
+                onValueChange = { onEmailChanged(it) },
+                label = { Text(text = "Email/Login") },
+                colors = TextFieldDefaults.textFieldColors(
+                        focusedIndicatorColor = Color.Transparent,
+                        unfocusedIndicatorColor = Color.Transparent
+                ),
+                shape = RoundedCornerShape(8.dp),
+                keyboardOptions = KeyboardOptions.Default.copy(keyboardType = KeyboardType.Email,
+                        imeAction = ImeAction.Next
+                ),
+                keyboardActions = KeyboardActions(
+                        onNext = {
+
+                            onImeAction()
+                        }
+                ),
+                isError = error != null
+        )
+        error?.let { ErrorField(it) }
+    }
+}
+@Composable
+fun ErrorField(error: String) {
+    Text(
+            text = error,
+            modifier = Modifier.fillMaxWidth(),
+            style = TextStyle(color = MaterialTheme.colors.error)
+    )
+}
+@Composable
+fun Password(password: String, error: String?, onPasswordChanged: (String) -> Unit) {
+    val showPassword = remember {
+        mutableStateOf(false)
+    }
+    TextField(
+            value = password,
+            onValueChange = {
+                onPasswordChanged(it)
+            },
+            label = { Text(text = "Password") },
+            colors = TextFieldDefaults.textFieldColors(
+                    focusedIndicatorColor = Color.Transparent,
+                    unfocusedIndicatorColor = Color.Transparent
+            ),
+            shape = RoundedCornerShape(8.dp),
+            visualTransformation = if (showPassword.value) {
+                VisualTransformation.None
+            } else {
+                PasswordVisualTransformation()
+            },
+            keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Password),
+            trailingIcon = {
+                if (showPassword.value) {
+                    IconButton(onClick = { showPassword.value = false }) {
+                        Icon(
+                                imageVector = Icons.Filled.Visibility,
+                                contentDescription = stringResource(R.string.hide_password)
+                        )
+                    }
+                } else {
+                    IconButton(onClick = { showPassword.value = true }) {
+                        Icon(
+                                imageVector = Icons.Filled.VisibilityOff,
+                                contentDescription = stringResource(R.string.show_password)
+                        )
+                    }
+                }
+            },
+            isError = error != null
+    )
+    error?.let { ErrorField(it) }
+}
+@Composable
+fun SignInButton(enabled: Boolean, navController:NavController) {
```

Comment:
Why do we need a separate component for just a simple Button? Use Button within `LoginPage.kt` composable, this composable seems a bit redundant.